### DISCUSSION TITLE
MemoryError on trying to send large files.

### DISCUSSION THREAD

**Discussion by MaxMyzer** [miguelgrinberg/microdot#169] [Q&A] [answered]
I am having a similar issue to #121, but the fix didn't work. The number it is attempting to allocate is `38400` but `44717` bytes. I am using a Raspberry Pi Pico W.  
Here is the full log:
```
Task exception wasn't retrieved
future: <Task> coro= <generator object 'serve' at 20012c70>
Traceback (most recent call last):
  File "uasyncio/core.py", line 1, in run_until_complete
  File "/lib/microdot_asyncio.py", line 269, in serve
  File "/lib/microdot_asyncio.py", line 337, in handle_request
  File "/lib/microdot_asyncio.py", line 158, in write
  File "uasyncio/stream.py", line 1, in stream_awrite
  File "uasyncio/stream.py", line 1, in write
MemoryError: memory allocation failed, allocating 38400 bytes
```

**Comment by miguelgrinberg** [comment]
Could you share the code of the route that triggers this error?

**Reply by MaxMyzer** [reply]
```
@app.route('/file.js')
async def index(request):
    with open("file.js", 'r', encoding='utf-8') as f:
        return f.read()
```
I have also now tried it on an ESP32.

**Reply by miguelgrinberg** [reply, marked as answer]
The issue is in your code, not in Microdot. Your `f.read()` call is bringing the entire file into memory. If you were to use the `send_file()` function from this framework, the file would be sent in small chunks, using a lot less memory.

**Reply by MaxMyzer** [reply]
Thank you! I was actually about to try doing that manually but couldn't figure it out, I don't use micropy a lot.
